File: c/string/word-count.c
#include <stdio.h>
int main(void)
{
    int i=0, words = 1;
    char sen[100];
    printf("Enter a sentence: ");
    scanf("%[^\n]", sen);
    // fgets(sen, 100, stdin);

    while (sen[i] != '\0')
    {
        if(sen[i] == ' ')
            words++;
        i++;
    }
    printf("Number of words: %d\n", words);
}

File: java/no-package/string/WordCount.java

import java.util.Scanner;

class WordCount {
    public static void main(String[] args) {
        int words = 1;
        Scanner s = new Scanner(System.in);
        System.out.print("Enter a sentence: ");
        String sen = s.nextLine();

        for(int i=0; i<sen.length(); i++) {
            if(sen.charAt(i) == ' ')
                words++;
        }
        System.out.println("Number of words: " + words);
    }
}